x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -243,9 +248,24 @@ protected LogicalPlan rule(Aggregate aggregate) {
             LogicalPlan plan = aggregate;
             if (changed) {
                 var source = aggregate.source();
-                plan = new Aggregate(aggregate.source(), aggregate.child(), aggregate.groupings(), newAggs);
+                if (newAggs.isEmpty() == false) {
+                    plan = new Aggregate(source, aggregate.child(), aggregate.groupings(), newAggs);
+                } else {
+                    // All aggs actually have been surrogates for (foldable) expressions, e.g.
+                    //
+                    // ...
+                    // | STATS a = AVG([1,2]), m = min([1,2,3])
+                    //
+                    // The output needs to be 1 row containing the results of the folding.
+                    // To achieve this, we replace the aggregation with a single row and let the subsequent Eval produce the values, so
+                    // the plan becomes essentially
+                    //
+                    // ROW placeholder = null
+                    // | EVAL a = MV_AVG([1,2]), m = MV_MIN([1,2,3])
+                    plan = emptyRow(source);
+                }
```

Comment:
@costin and I discussed another, similar approach using `LocalRelation`, namely replacing
```
... | STATS a = AVG([1,2]), m = min([1,2,3])
```
by
```
LocalRelation(source, attributes, singleRowLocalSupplier)`
```
where `attributes` consists of `a` and `m`, and `singleRowLocalSupplier` has a single row containing the values `MV_AVG([1,2])` and `MV_MIN([1,2,3])` (folded).

I don't think this is the best approach for the following reasons:
- Creating the `singleRowLocalSupplier` requires manually creating blocks with the folded values - this means that in addition to our `ConstantFolding` optimizer rule, we have another place that does constant folding; and we need to manually create proper blocks this time rather than literals, meaning that more low-level block handling leaks into the planner code.
- This does not benefit from optimization rules that prune unused `Eval` expressions, while the present approach does.
- This opens up 1 special edge case in which we do not use an `Eval` to compute the values of the surrogate expressions but need to do this manually; this may require additional edge case handling later on.